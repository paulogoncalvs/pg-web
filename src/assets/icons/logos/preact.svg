<svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 256 296"><path d="M128 0L0 74v148l128 74 128-74V74L128 0zm68 66c11 0 19 3 24 10 12 15 2 43-23 72 25 29 35 57 23 72-14 18-51 11-92-14-41 25-79 32-93 14-12-15-1-43 23-72-24-29-35-57-23-72 5-7 14-10 25-10 18 0 42 9 68 24 25-15 50-24 68-24zm0 9c-7 0-14 2-23 4-11 4-24 9-37 17a298 298 0 0155 45l14-20c11-18 14-32 8-40-3-4-9-6-17-6zM59 75c-8 0-14 2-17 6-6 8-3 22 8 40l15 20a282 282 0 0154-45c-13-8-26-13-37-17-9-2-16-4-23-4zm69 26a284 284 0 00-57 47 272 272 0 0057 47 283 283 0 0056-47 272 272 0 00-56-47zm0 27a20 20 0 010 40 20 20 0 010-40zm-63 27l-15 21c-11 17-14 32-8 39s20 8 40 2c11-4 24-9 37-16a298 298 0 01-54-46zm126 0a282 282 0 01-55 46c13 7 26 12 37 16 20 6 34 5 40-2s3-22-8-39l-14-21z"/></svg>