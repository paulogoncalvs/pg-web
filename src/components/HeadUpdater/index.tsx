import routesConfig from '@/config/routes/index.js';
import { useRouter } from '@/modules/router';
import { isBrowser } from '@/utils/browser';

export const HeadUpdater = (): null => {
    useRouter();

    if (!isBrowser()) {
        return null;
    }

    const data = routesConfig[window.location.pathname] || {};

    if (data.templateParameters) {
        const { title } = data.templateParameters.head;

        document.title = title;
    }

    return null;
};
