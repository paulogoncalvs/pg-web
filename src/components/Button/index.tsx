import classNames from 'classnames';
import { h, FunctionalComponent, ComponentChildren } from 'preact';

interface ButtonComponentProps {
    mainClasses?: string;
    classes?: string;
    otherProps?: unknown;
    children?: ComponentChildren;
}

export const Button: FunctionalComponent<ButtonComponentProps> = ({
    mainClasses = 'btn',
    classes,
    children,
    ...otherProps
}) => (
    <button class={classNames(mainClasses, classes)} {...otherProps}>
        {children}
    </button>
);
