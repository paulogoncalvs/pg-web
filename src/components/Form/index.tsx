import { h, FunctionalComponent, ComponentChildren, createContext } from 'preact';
import { useState } from 'preact/hooks';

export interface FormComponentData {
    [key: string]: string;
}
interface FormComponentProps {
    action: string;
    initialValues: FormComponentData;
    onSubmit(formData: FormComponentData): void;
    classes: string;
    children: ComponentChildren;
}

export const FormContext = createContext({
    formData: {},
    handleFormChange: (event: Event): void => console.warn(event), // eslint-disable-line no-console
});

export const Form: FunctionalComponent<FormComponentProps> = ({
    action,
    initialValues,
    onSubmit,
    classes,
    children,
}) => {
    const [formData, setFormData] = useState(initialValues);

    const handleFormChange = (event: Event): void => {
        const { name, value } = event.target as HTMLInputElement;

        setFormData({ ...formData, [name]: value });
    };

    const triggerSubmit = (event: Event): void => {
        if (!onSubmit) {
            return;
        }

        event.preventDefault();
        onSubmit(formData);
    };

    return (
        <form action={action} onSubmit={triggerSubmit} class={classes}>
            <FormContext.Provider value={{ formData, handleFormChange }}>{children}</FormContext.Provider>
        </form>
    );
};
