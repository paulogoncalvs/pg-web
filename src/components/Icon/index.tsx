import { h, FunctionalComponent } from 'preact';
import { useContext } from 'preact/hooks';
import { StoreContext } from '@/modules/store';

interface IconComponentProps {
    src: IconSrc;
    width?: string;
    height?: string;
    classes?: string;
    viewBox?: string;
    ariaHidden?: boolean;
    otherProps?: unknown;
}

export const Icon: FunctionalComponent<IconComponentProps> = ({
    src: [hash = '#', viewBox] = [],
    width = '24',
    height = '24',
    classes = 'fill-current',
    ariaHidden,
    ...otherProps
}) => {
    const { filenames } = useContext(StoreContext);

    return (
        <svg
            {...otherProps}
            aria-hidden={ariaHidden}
            class={classes}
            width={width}
            height={height}
            {...(viewBox ? { viewBox } : {})}
        >
            <use href={`${filenames ? filenames.sprite : ''}${hash}`} />
        </svg>
    );
};
