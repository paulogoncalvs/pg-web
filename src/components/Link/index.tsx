import { h, FunctionalComponent, ComponentChildren } from 'preact';
import { useTranslate } from '@/modules/i18n';

interface LinkComponentProps {
    href?: string;
    class?: string;
    newWindow?: boolean;
    ariaLabel?: string;
    onClick?(): void;
    children?: ComponentChildren;
}

export const Link: FunctionalComponent<LinkComponentProps> = ({
    href,
    newWindow,
    ariaLabel,
    children,
    ...otherProps
}) => {
    const { t } = useTranslate();

    return (
        // @ts-ignore-begin
        <a
            href={href}
            aria-label={
                ariaLabel ? (newWindow ? t('accessibility_new_window', { text: ariaLabel }) : ariaLabel) : undefined
            }
            {...(newWindow ? { target: '_blank', rel: 'noopener noreferrer' } : {})}
            {...otherProps}
        >
            {children}
        </a>
        // @ts-ignore-end
    );
};
