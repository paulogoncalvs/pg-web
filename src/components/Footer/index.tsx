import { h, FunctionalComponent } from 'preact';
import { Icon } from '@/components/Icon';
import { Link } from '@/components/Link';
import { t } from '@/modules/i18n';
import preactLogo from '@/assets/icons/logos/preact.svg';
import webpackLogo from '@/assets/icons/logos/webpack.svg';
import tailwindLogo from '@/assets/icons/logos/tailwind.svg';
import babelLogo from '@/assets/icons/logos/babel.svg';
import esLintLogo from '@/assets/icons/logos/eslint.svg';
import postCSSLogo from '@/assets/icons/logos/postcss.svg';
import prettierLogo from '@/assets/icons/logos/prettier.svg';
import sassLogo from '@/assets/icons/logos/sass.svg';
import styleLintLogo from '@/assets/icons/logos/stylelint.svg';
import typeScriptLogo from '@/assets/icons/logos/typescript.svg';
import yarnLogo from '@/assets/icons/logos/yarn.svg';
import workboxLogo from '@/assets/icons/logos/workbox.svg';
import jestLogo from '@/assets/icons/logos/jest.svg';

const initialYear = 2021;
const currentYear = new Date().getFullYear();

export const Footer: FunctionalComponent = () => (
    <footer>
        <div class="bg-gray-100 dark:bg-gray-900 text-center">
            <div class="container mx-auto px-6 pt-14 pb-12 flex flex-col items-center">
                <div class="pb-6 text-gray-600 dark:text-gray-400 text-sm sm:text-base">
                    {t('footer_description_1')}
                </div>
                <div class="flex flex-wrap justify-center align-middle">
                    <Link href="https://www.typescriptlang.org/" classes="ic-link _ftr" newWindow>
                        <Icon src={typeScriptLogo} aria-label="TypeScript" width="32" height="32" />
                    </Link>
                    <Link href="https://preactjs.com/" classes="ic-link _ftr" newWindow>
                        <Icon src={preactLogo} aria-label="Preact" width="32" height="32" />
                    </Link>
                    <Link href="https://webpack.js.org/" classes="ic-link _ftr" newWindow>
                        <Icon src={webpackLogo} aria-label="Webpack" width="32" height="32" />
                    </Link>
                    <Link href="https://yarnpkg.com/" classes="ic-link _ftr" newWindow>
                        <Icon src={yarnLogo} aria-label="Yarn" width="32" height="32" />
                    </Link>
                    <Link href="https://eslint.org/" classes="ic-link _ftr" newWindow>
                        <Icon src={esLintLogo} aria-label="ESLint" width="32" height="32" />
                    </Link>
                    <Link href="https://stylelint.io/" classes="ic-link _ftr" newWindow>
                        <Icon src={styleLintLogo} aria-label="StyleLint" width="32" height="32" />
                    </Link>
                    <Link href="https://developers.google.com/web/tools/workbox" classes="ic-link _ftr" newWindow>
                        <Icon src={workboxLogo} aria-label="Workbox" width="32" height="32" />
                    </Link>
                    <Link href="https://tailwindcss.com/" classes="ic-link _ftr" newWindow>
                        <Icon src={tailwindLogo} aria-label="Tailwind CSS" width="32" height="32" />
                    </Link>
                    <Link href="https://prettier.io/" classes="ic-link _ftr" newWindow>
                        <Icon src={prettierLogo} aria-label="Prettier" width="32" height="32" />
                    </Link>
                    <Link href="https://jestjs.io/" classes="ic-link _ftr" newWindow>
                        <Icon src={jestLogo} aria-label="Jest" width="32" height="32" />
                    </Link>
                    <Link href="https://postcss.org/" classes="ic-link _ftr" newWindow>
                        <Icon src={postCSSLogo} aria-label="PostCSS" width="32" height="32" />
                    </Link>
                    <Link href="https://sass-lang.com/" classes="ic-link _ftr" newWindow>
                        <Icon src={sassLogo} aria-label="Sass" width="32" height="32" />
                    </Link>
                    <Link href="https://babeljs.io/" classes="ic-link _ftr" newWindow>
                        <Icon src={babelLogo} aria-label="Babel" width="60" height="32" />
                    </Link>
                </div>
            </div>
        </div>
        <div class="container mx-auto py-8 flex flex-col items-center text-center text-gray-500 dark:text-gray-400 text-xs">
            <p>{t('footer_description_2')}</p>
            <p class="pt-2">
                paulogoncalves.dev &copy; {initialYear} {currentYear > initialYear ? ` - ${currentYear}` : ''} ðŸ™‚
            </p>
        </div>
    </footer>
);
