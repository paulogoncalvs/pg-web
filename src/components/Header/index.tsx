import { h, FunctionalComponent } from 'preact';
import { trackEvent } from '@/modules/tracking';
import { ToggleTheme } from '@/components/ToggleTheme';
import { Icon } from '@/components/Icon';
import { Link } from '@/components/Link';
import LanguageSelector from '@/components/LanguageSelector';
import linkedInIcon from '@/assets/icons/logos/linkedin.svg';
import githubIcon from '@/assets/icons/logos/github.svg';
import twitterIcon from '@/assets/icons/logos/twitter.svg';

export const Header: FunctionalComponent = () => (
    <header class="container p-4 py-10 mx-auto">
        <div class="flex items-center justify-between">
            <div class="flex items-center">
                <LanguageSelector classes="ml-2 mr-2" />
            </div>
            <div class="flex justify-end">
                <ToggleTheme classes="p-2" />
                <Link
                    href="https://www.twitter.com/paulogoncalvs"
                    class="ic-link _hdr"
                    ariaLabel="Twitter"
                    newWindow
                    onClick={(): void =>
                        trackEvent({
                            eventCategory: 'Header Link',
                            eventAction: 'clickLink',
                            eventLabel: 'Twitter',
                        })
                    }
                >
                    <Icon src={twitterIcon} ariaHidden />
                </Link>
                <Link
                    href="https://www.github.com/paulogoncalvs"
                    class="ic-link _hdr"
                    ariaLabel="GitHub"
                    newWindow
                    onClick={(): void =>
                        trackEvent({
                            eventCategory: 'Header Link',
                            eventAction: 'clickLink',
                            eventLabel: 'GitHub',
                        })
                    }
                >
                    <Icon src={githubIcon} ariaHidden />
                </Link>
                <Link
                    href="https://pt.linkedin.com/in/paulogoncalvs"
                    class="ic-link _hdr"
                    ariaLabel="LinkedIn"
                    newWindow
                    onClick={(): void =>
                        trackEvent({
                            eventCategory: 'Header Link',
                            eventAction: 'clickLink',
                            eventLabel: 'LinkedIn',
                        })
                    }
                >
                    <Icon src={linkedInIcon} ariaHidden />
                </Link>
            </div>
        </div>
    </header>
);
