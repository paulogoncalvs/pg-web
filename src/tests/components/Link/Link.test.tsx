import { h } from 'preact';
import { render } from 'preact-render-to-string';
import { Link } from '@/components/Link';

describe('[COMPONENT] Link Component', () => {
    it('should render default', () => {
        expect(render(<Link href="/test/" />)).toMatchSnapshot();
    });

    it('should render to open on a new window', () => {
        expect(render(<Link href="/test/" newWindow />)).toMatchSnapshot();
    });
});
