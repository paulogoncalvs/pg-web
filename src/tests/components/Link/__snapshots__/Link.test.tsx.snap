// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[COMPONENT] Link Component should render default 1`] = `
<a href="/test/">
</a>
`;

exports[`[COMPONENT] Link Component should render to open on a new window 1`] = `
<a href="/test/"
   target="_blank"
   rel="noopener noreferrer"
>
</a>
`;
